### PR TITLE
Feature/backend/register user endpoint

### PR DESCRIPTION
**Pull request by Jorgro**
## Related Issue
Closes https://trello.com/c/awTp3dGw/29-improve-endpoints-for-user-api

## Proposed changes
- Adds `isAdmin` field to JWT token and a `verifyAdminToken` for endpoints where the user must be admin. 
- Added endpoint for adding new users (checks if username is already taken)

## How has this been tested?
- [x] Tested the endpoint both with admin and non-admin user. 

## Checklist
- [x] Tests
- [x] Documentation

## Screenshots (if appropriate):

### CODE REVIEW

File: backend/src/auth/credentials.ts [hunk 1]
```diff
@@ -76,6 +76,19 @@ export const checkPassword = (password: string, existingHash: string) => {
   return diff === 0;
 };
 
+export const verifyAdminToken = (token: string) => {
+  try {
+    const { exp } = jwt.verify(token, config.JWT_SECRET_TOKEN) as { exp };
+    const { isAdmin } = jwt.decode(token, config.JWT_SECRET_TOKEN);
```
Comment (espandre) [inline backend/src/auth/credentials.ts:82]: You should be able to fold this into the first verify call, as we're now decoding the token twice.

File: backend/src/database/queries/setUser.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  return `
+    ${prefixString}
+    insert {
+      ?uri rdf:type SDG:User .
+      ?uri SDG:username "${username}".
+      ?uri SDG:passwordHash "${password}".
+      ?uri SDG:userHasRole ?role.
+   }
+   where {
+    BIND(IRI("http://www.semanticweb.org/aga/ontologies/2017/9/SDG#datapoint.u4ssc.${username}") as ?uri)
```
Comment (espandre) [inline backend/src/database/queries/setUser.ts:15]: I think that a different IRI-path would be more suitable, eg. `"http://www.semanticweb.org/aga/ontologies/2017/9/SDG#user.${username}"`.

File: backend/src/database/queries/setUser.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    BIND(IRI("${role}") as ?role)
```
Comment (espandre) [inline backend/src/database/queries/setUser.ts:16]: Similarly, the existing roles are `sdg.userrole.admin` and `sdg.userrole.dataentry`, such that this string should be "http://www.semanticweb.org/aga/ontologies/2017/9/SDG#sdg.userrole.${role}"

File: utils/data_generator/datagen.py [hunk 4]
```diff
@@ -46,6 +46,20 @@ def insert_data(token, kpi, value, municipality, year, dataseries=None):
     return json.loads(req.text)
 
 
+def insert_user(token, username, password, role):
```
Comment (espandre) [inline utils/data_generator/datagen.py:49]: This should not be in the data generator, the purpose of which is to generate datapoints for the data entry api and goal distance computation. Should be part of the `make_user` utility.

File: backend/src/database/queries/setUser.ts (hunk 2 above)
Comment (Jorgro) [reply backend/src/database/queries/setUser.ts:15]: Haha ofc, copied from another query and forgot to replace 😓

**Review by espandre — APPROVED** [review]
LGTM.

**Pull request merged**